Popup docs (#1877)
telerik/blazor-docs: components/popup/position-collision.md

---
title: Position and Collision
page_title: Popup Position and Collision
description: Discover the placement settings of the Popup for Blazor. Learn how to configure the Popup position and handle collisions.
slug: popup-position-collision
tags: telerik,blazor,popup,position,collision
published: True
position: 10
---

> # Popup Position and Collision Settings

Reviewer: We don't use "Settings" or similar words in titles.

```suggestion
# Popup Position and Collision
```

> This article outlines the available settings which enable you to control the position of the Popup based on its anchor and dictate how the component responds to insufficient screen space.

Reviewer 2: ```suggestion
This article outlines how to control the position of the Popup based on its anchor and dictate how the component responds to insufficient screen space.
```

## Position

You can customize how the popup and its anchor align with each other. You can use the available parameters to control the position of the component.

### Anchor Alignment

> Use the available parameters to control which part of the Popup touches the anchor.

Reviewer: ```suggestion
Use the available parameters to control which part of the anchor touches the Popup.
```

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Parameter | Type | Enum Members |
| ----------- | ----------- | ----------- |
| `AnchorHorizontalAlign` | `PopupAnchorHorizontalAlign` | `Center` <br /> `Left` (default value) <br /> `Right` |
| `AnchorVerticalAlign` | `PopupAnchorVerticalAlign` | `Bottom` (default value) <br /> `Center` <br /> `Top` |

See the [example](#example) below to customize the available parameters and see how they affect the Popup component.

### Popup Alignment

Use the available parameters to control which part of the anchor touches the popup.

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Parameter | Type | Enum Members |
| ----------- | ----------- | ----------- |

> | `HorizontalAlign` | `PopupHorizontalAlign` | `Center` <br /> `Left` (default value) <br /> `Right` |

Author: Yes, this is part of the enum members

> | `HorizontalAlign` | `PopupHorizontalAlign` | `Center` <br /> `Left` (default value) <br /> `Right` |

Reviewer 2: Consider removing the extra spaces around your `<br />`s throughout the table.

| `VerticalAlign` | `PopupVerticalAlign` | `Bottom` <br /> `Center` <br /> `Top` (default value) |

## Collision

There are two collision modes available for the Popup component:

> * `Fit`&mdash;The Popup will be shifted until it is fully visible on the screen.

Reviewer: ```suggestion
* `Fit`&mdash;The Popup will shift until it is fully visible on the screen.
```

* `Flip` - The Popup will render on the other side of the anchor.

Use the available parameters to control how the Popup behaves to insufficient screen space.

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Parameter | Type | Enum Members |
| ----------- | ----------- | ----------- |
| `HorizontalCollision` | `PopupCollision ` | `Fit` (default value) <br /> `Flip`  |
| `VerticalCollision` | `PopupCollision ` | `Fit` <br /> `Flip` (default value) |

See the [example](#example) below to customize the available parameters and see how they affect the Popup component.

## Example

> The following example lets you experiment with the available settings that control the position and collision behavior in the Popup. It starts with the default component behavior.

Reviewer 2: ```suggestion
The following example lets you experiment with the available settings that control the position and collision behavior of the Popup. It starts with the default component behavior.
```

        PopupAnchorHorizontalAlign.Left,
        PopupAnchorHorizontalAlign.Center,
        PopupAnchorHorizontalAlign.Right
    };

    private readonly List<PopupAnchorVerticalAlign> PopupAnchorVerticalAligns = new List<PopupAnchorVerticalAlign>() {
        PopupAnchorVerticalAlign.Bottom,
        PopupAnchorVerticalAlign.Center,
        PopupAnchorVerticalAlign.Top
    };

    private readonly List<PopupHorizontalAlign> PopupHorizontalAligns = new List<PopupHorizontalAlign>() {
        PopupHorizontalAlign.Left,
        PopupHorizontalAlign.Center,
        PopupHorizontalAlign.Right
    };

    private readonly List<PopupVerticalAlign> PopupVerticalAligns = new List<PopupVerticalAlign>() {
        PopupVerticalAlign.Bottom,
        PopupVerticalAlign.Center,
        PopupVerticalAlign.Top
    };

    private readonly List<PopupCollision> PopupCollisions = new List<PopupCollision>() {
        PopupCollision.Fit,
        PopupCollision.Flip
    };
}

<style>
    .styled-container {
        width: 200px;
        height: 30px;
        background-color: yellowgreen;
        margin-top: 20px;
    }
</style>

> ````

Reviewer 2: Consider adding a See Also section here as well. For example:

```
## See Also

* [Popup API Reference](/blazor-ui/api/Telerik.Blazor.Components.TelerikPopup)
* [Popup Overview]({%slug popup-overview%})
* [Live Popup Demos](https://demos.telerik.com/blazor-ui/popup/overview)
```
